File: macroslib/tests/expectations/access.cpp
r#"private:

    FooWrapper(int32_t a0) noexcept
    {

        this->self_ = private_Foo_from_int(a0);
        if (this->self_ == nullptr) {
            std::abort();
        }
    }

    static void private_f() noexcept;
public:

    static void public_f() noexcept;
protected:

    static void protected_f() noexcept;"#;

File: macroslib/tests/expectations/access.rs
foreign_class!(
    class Foo {
        self_type Foo;
        constructor Foo::new() -> Foo;
        private constructor Foo::from_int(_: i32) -> Foo;
        private fn Foo::private_f();
        fn Foo::public_f();
        protected fn Foo::protected_f();
    }
);
